File: edX/Harvard CS50/pset6/readability/readability.py
from cs50 import get_string


def main():
    Input = get_string("Text: ")

    Length = len(Input)

    letters = 0
    words = 0
    sentences = 0

    for i in range(Length):
        if str.isalpha(Input[i]) == True:
            letters += 1
        elif Input[i] == "!" or Input[i] == "." or Input[i] == "?":
            sentences += 1
        elif Input[i] == " ":
            words += 1

    words += 1

    level = round(index((letters / float(words) * 100), sentences / float(words) * 100))

    if level < 1:
        print("Before Grade 1")
    elif level > 16:
        print("Grade 16+")
    else:
        print(f"Grade {level}")

    return


def index(L, S):

    answer = float(0.0588 * L) - (0.296 * S) - 15.8
    return answer


main()

File: edX/Harvard CS50/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

//Declaring function at the start
float index(float L, float S);

int main(void)
{
    //Declaring variables, inputs, etc.
    string input;

    //Prompting input
    input = get_string("Please input the thing here: ");

    //Getting the string length
    int length = strlen(input);

    //Declaring essential variables
    int letters = 0;
    int words = 0;
    int sentences = 0;


    //Using arrays to obtain individual chars of the string input
    for (int i = 0; i < length; i++)
    {
        if (isalpha(input[i]))
        {
            letters++;
        }
        else if (input[i] == '!' || input[i] == '.' || input[i] == '?')
        {
            sentences++;
        }
        else if (input[i] == ' ')
        {
            words++;
        }

    }

    //Counting spaces for words require 1 more space in the end
    words++;

    //Calculating the grade that is needed
    int level = round(index((letters / (float) words) * 100, (sentences / (float) words) * 100));

    //Debugging used codes
    // printf("Letters: %i\n", letters);
    // printf("Words: %i\n", words);
    // printf("Sentences: %i\n", sentences);
    // printf("Index: %i\n", level);

    //Final printing in the answers
    if (level < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (level > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", level);
    }

}

//The function for index
float index(float L, float S)
{
    float answer = (0.0588 * L) - (0.296 * S) - 15.8;
    return answer;
}